.github/workflows/main.yml

```diff
@@ -34,6 +34,14 @@ jobs:
           zenml init --analytics_opt_in False
           python ../smoke_test.py && echo "Finished smoke test successfully" || echo "Smoke test failed"
 
+      - name: unit-tests
```

Comment:
I would now use the testing.sh script here. Lets try it out

Reply:
Good point! I changed it to run the script directly in 81383b9

Reply:
Have you tried testing it out? :-) Just a small test before we merge

Reply:
Nope, have not tested it since it's a Github workflow, which I have no idea how to run locally. We could maybe test it with a manual workflow dispatch trigger

Reply:
Yes lets do that then first

Reply:
Fixed in 55831eb and b983baf , now the GitHub action fails whenever there are failed tests, and succeeds if and only if all tests pass. Analytics opt out is given, zenml init is never called, smoke test block was removed because the unit tests contain a separate smoke test in `generate_test_pipelines.py`.

b9351e3 fixes the schema issue that previously failed the datasource equality test.